src/dstack/_internal/server/models.py

```diff
@@ -230,3 +241,76 @@ class GatewayComputeModel(BaseModel):
     ssh_public_key: Mapped[str] = mapped_column(Text)
 
     deleted: Mapped[bool] = mapped_column(Boolean, server_default=false())
+
+
+class PoolModel(BaseModel):
+    __tablename__ = "pools"
+
+    id: Mapped[uuid.UUID] = mapped_column(
+        UUIDType(binary=False), primary_key=True, default=uuid.uuid4
+    )
+    name: Mapped[str] = mapped_column(String(50))
+    created_at: Mapped[datetime] = mapped_column(DateTime, default=get_current_datetime)
+    deleted: Mapped[bool] = mapped_column(Boolean, default=False)
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(DateTime)
+
+    project_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("projects.id", ondelete="CASCADE"))
+    project: Mapped["ProjectModel"] = relationship(foreign_keys=[project_id])
+
+    instances: Mapped[List["InstanceModel"]] = relationship(back_populates="pool", lazy="selectin")
+
+
+class InstanceModel(BaseModel):
+    __tablename__ = "instances"
+
+    id: Mapped[uuid.UUID] = mapped_column(
+        UUIDType(binary=False), primary_key=True, default=uuid.uuid4
+    )
+    name: Mapped[str] = mapped_column(String(50))
+    created_at: Mapped[datetime] = mapped_column(DateTime, default=get_current_datetime)
+    deleted: Mapped[bool] = mapped_column(Boolean, default=False)
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(DateTime)
+
+    project_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("projects.id", ondelete="CASCADE"))
+    project: Mapped["ProjectModel"] = relationship(foreign_keys=[project_id], single_parent=True)
+
+    pool_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("pools.id"))
+    pool: Mapped["PoolModel"] = relationship(back_populates="instances", single_parent=True)
+
+    status: Mapped[InstanceStatus] = mapped_column(Enum(InstanceStatus))
+    status_message: Mapped[Optional[str]] = mapped_column(String(50))
+
+    # VM
+    started_at: Mapped[Optional[datetime]] = mapped_column(DateTime, default=get_current_datetime)
+    finished_at: Mapped[Optional[datetime]] = mapped_column(DateTime, default=get_current_datetime)
+
+    termination_policy: Mapped[Optional[TerminationPolicy]] = mapped_column(String(50))
+    termination_idle_time: Mapped[Optional[str]] = mapped_column(String(50))
+
+    backend: Mapped[BackendType] = mapped_column(Enum(BackendType))
+    backend_data: Mapped[Optional[str]] = mapped_column(String(4000))
+    region: Mapped[str] = mapped_column(String(2000))
+    price: Mapped[float] = mapped_column(Float)
+
+    job_provisioning_data: Mapped[str] = mapped_column(String(4000))
+
+    offer: Mapped[str] = mapped_column(String(4000))
+
+    resource_spec_data: Mapped[Optional[str]] = mapped_column(String(4000))
```

Comment:
Unclear why having optional resource_spec_data. Remove if not needed.

Reply:
Fixed.